go-controller/pkg/node/node_ip_handler_linux_test.go

```diff
@@ -198,3 +151,88 @@ var _ = Describe("Node IP Handler tests", func() {
 		})
 	})
 })
+
+func configureKubeOVNContext(nodeName string, useNetlink, createNs, createLink bool) *testCtx {
+	node := &corev1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: nodeName,
+			Annotations: map[string]string{
+				util.OVNNodeHostCIDRs:           `["10.1.1.10/24", "2001:db8::10/64"]`,
+				"k8s.ovn.org/l3-gateway-config": `{"default":{"mac-address":"52:54:00:e2:ed:d0","ip-addresses":["10.1.1.10/24"],"ip-address":"10.1.1.10/24","next-hops":["10.1.1.1"],"next-hop":"10.1.1.1"}}`,
+			},
+		},
+		Status: corev1.NodeStatus{
+			Addresses: []corev1.NodeAddress{{Address: "10.1.1.10", Type: corev1.NodeInternalIP}},
+		},
+	}
+
+	tc := &testCtx{
+		doneWg:      &sync.WaitGroup{},
+		stopCh:      make(chan struct{}),
+		fakeClient:  fake.NewSimpleClientset(node),
+		mgmtPortIP4: ovntest.MustParseIPNet("10.1.1.2/24"),
+		mgmtPortIP6: ovntest.MustParseIPNet("2001:db8::1/64"),
+	}
+
+	var err error
+	fakeClientset := &util.OVNNodeClientset{
+		KubeClient: tc.fakeClient,
+	}
+	tc.watchFactory, err = factory.NewNodeWatchFactory(fakeClientset, nodeName)
+	Expect(err).NotTo(HaveOccurred())
+	err = tc.watchFactory.Start()
+	Expect(err).NotTo(HaveOccurred())
+
+	fakeMgmtPortConfig := &managementPortConfig{
+		ifName:    nodeName,
+		link:      nil,
+		routerMAC: nil,
+		ipv4: &managementPortIPFamilyConfig{
+			ipt:        nil,
+			allSubnets: nil,
+			ifAddr:     tc.mgmtPortIP4,
+			gwIP:       tc.mgmtPortIP4.IP,
+		},
+		ipv6: &managementPortIPFamilyConfig{
+			ipt:        nil,
+			allSubnets: nil,
+			ifAddr:     tc.mgmtPortIP6,
+			gwIP:       tc.mgmtPortIP6.IP,
+		},
+	}
+
+	fakeBridgeConfiguration := &bridgeConfiguration{bridgeName: "breth0"}
+
+	k := &kube.Kube{KClient: tc.fakeClient}
+	if createNs {
```

Comment:
so I would just make this another util:
`configureKubeOVNContextWithNs` which calls `configureKubeOVNContext` and then adds the extra plumbing you need like replace the ipManager or whatever..
then use configureKubeOVNContext from the old test and configureKubeOVNContextWithNs from your new test in next commit

Reply:
Ya, thats cleaner. Thanks!